Source/SwiftLintFramework/Rules/Lint/DuplicateConditionsRule.swift

```diff
@@ -0,0 +1,214 @@
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension DuplicateConditionsRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ node: IfStmtSyntax) {
+            if let prevToken = node.previousToken,
+               prevToken.tokenKind == .elseKeyword {
+                // We can skip these cases - they will be picked up when we visit the top level `if`
+                return
+            }
+
+            var maybeCurr: IfStmtSyntax? = node
+            var statementChain: [IfStmtSyntax] = []
+            while let curr = maybeCurr {
+                statementChain.append(curr)
+                maybeCurr = curr.elseBody?.as(IfStmtSyntax.self)
+            }
+
+            let positionsByConditions = statementChain
+                .reduce(into: [Set<[UInt8]>: [AbsolutePosition]]()) { acc, elt in
+                    let conditions = elt.conditions.compactMap(extract)
+                    let location = elt.conditions.positionAfterSkippingLeadingTrivia
+                    acc[Set(conditions), default: []].append(location)
+                }
+
+            addViolations(positionsByConditions)
+        }
+
+        override func visitPost(_ node: SwitchCaseListSyntax) {
+            let switchCases = node.compactMap { $0.as(SwitchCaseSyntax.self) }
+
+            let positionsByCondition = switchCases
+                .reduce(into: [[UInt8]: [AbsolutePosition]]()) { acc, elt in
+                    // Defaults don't have a condition to worry about
+                    guard case let .case(caseLabel) = elt.label else { return }
+                    for caseItem in caseLabel.caseItems {
+                        let pattern = caseItem.pattern.withoutTrivia().syntaxTextBytes
+                        let whereClause = caseItem.whereClause?.withoutTrivia().syntaxTextBytes ?? []
+                        let location = caseItem.positionAfterSkippingLeadingTrivia
+                        acc[pattern + whereClause, default: []].append(location)
+                    }
+                }
+
+            addViolations(positionsByCondition)
+        }
+
+        private func extract(_ node: ConditionElementSyntax) -> [UInt8]? {
+            let text: [UInt8]
+            switch node.condition {
+            case .availability(let node):
+                text = node.withoutTrivia().syntaxTextBytes
+            case .expression(let node):
+                text = node.withoutTrivia().syntaxTextBytes
+            case .matchingPattern(let node):
+                text = node.withoutTrivia().syntaxTextBytes
+            case .optionalBinding(let node):
+                text = node.withoutTrivia().syntaxTextBytes
+            default:
+                return nil
+            }
+
+            return text
+        }
+
+        private func addViolations<T>(_ positionsByCondition: [T: [AbsolutePosition]]) {
```

Comment:
Would be enough to pass a `[[AbsolutePosition]]` in here.

Reply:
Generic dropped by moving to `any Collection` (using an Array would mean wrapping `.values` in an Array initialiser)

Reply:
Ha, frustratingly, Jazzy is freaking out over the use of `any` or `some` when trying to generate the docs:
```
error: cannot specialize protocol type 'Collection'
```

`[[AbsolutePosition]]`` it is :+1: